### PR TITLE
Add experimental microcontroller mixin

### PR DESCRIPTION
**Pull request by nhunzaker**
This is the beginning of a ["getters"](https://github.com/optimizely/nuclear-js#performance) or ["selectors"](https://github.com/faassen/reselect) like feature for Microcosm. It will probably never be as robust as NuclearJS's getters, which can leverage ImmutableJS hash codes, but it can at least aspire to improve efficiency.

This is just the subscription system. For actual selection, we could actually use https://github.com/faassen/reselect

### CODE REVIEW

File: addons/microcontroller/index.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
+/**
+ * Microcontroller
```
Comment (ltk) [inline addons/microcontroller/index.js]: I think you could go for a more descriptive name for this. controller is already a kind of loaded word in software.

Comment (nhunzaker) [inline addons/microcontroller/index.js]: Agreed. I'd actually like something much more concise. Let's brainstorm!

File: Makefile [hunk 2]
```diff
@@ -33,6 +33,9 @@ javascript-min: javascript
 release: clean build
 	npm publish $(DIST)
 
+release-beta: clean build
```
Comment (greypants) [inline Makefile]: ooh. these are handy.

File: addons/microcontroller/index.js (hunk 1 above)
Comment (greypants) [inline addons/microcontroller/index.js]: Why not `ComputedProperties`? Also, I feel like `selections` should also be `computedProperties`. It better describes them, and would link it to the mixin, so it's clear they go with each other.

**Comment by greypants** [comment]
:+1:

File: addons/microcontroller/index.js (hunk 1 above)
Comment (averyvery) [inline addons/microcontroller/index.js]: `helpComputer`

`stop all the downloading`

File: addons/microcontroller/index.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  },
+
+  componentDidMount() {
+    const instance = this._getInstance()
+
+    if (!instance) {
+      throw TypeError(`Microcontroller expected ${this.constructor.displayName} to have an "app" or "instance" property but got ${ typeof instance }`)
+    }
+
+    this._getInstance().listen(this._change)
```
Comment (averyvery) [inline addons/microcontroller/index.js]: Since you already have instance, why use `_getInstance` again?

**Comment by averyvery** [comment]
:+1:

File: addons/microcontroller/index.js (hunk 3 above)
Comment (nhunzaker) [inline addons/microcontroller/index.js]: Reasons, Doug. Reasons... 

I'll fix it

**Comment by nhunzaker** [comment]
Great feedback. Closing for now until the name thing can be sorted out.

**Pull request closed without merging**